apis/config/v1alpha1/clientconfig_types.go

```diff
@@ -94,8 +98,88 @@ type ClientOptions struct {
 type CLIOptions struct {
 	// Repositories are the plugin repositories.
 	Repositories []PluginRepository `json:"repositories,omitempty" yaml:"repositories"`
+	// DiscoverySources determines from where to discover standalone plugins
+	DiscoverySources []PluginDiscovery `json:"discoverySources,omitempty" yaml:"discoverySources"`
 	// UnstableVersionSelector determined which version tags are allowed
 	UnstableVersionSelector VersionSelectorLevel `json:"unstableVersionSelector,omitempty" yaml:"unstableVersionSelector"`
+	// UseContextAwareDiscovery determines whether to use legacy way of discovering plugins or
+	// to use new Plugin API based plugin descovery
```

Comment:
Nit: mentioning context-aware will make it clearer that setting to true refers to the latter, not the former

```suggestion
	// to use the new context-aware Plugin API based plugin discovery mechanism
```
or something like that.